packages/flutter_tools/lib/src/base/config.dart

```diff
@@ -21,10 +21,19 @@ class Config {
     @required Logger logger,
     @required Platform platform,
   }) {
-    final File file = fileSystem.file(fileSystem.path.join(
+    File file = fileSystem.file(fileSystem.path.join(
       _userHomePath(platform),
-      name,
+      '.flutter_$name'
     ));
+    if (!file.existsSync()) {
+      final FileSystemUtils fsUtils = FileSystemUtils(platform: platform, fileSystem: fileSystem);
```

Comment:
Instead of using FSutils here I would just do:

```
file.parent.createSync(recursive: true);
```

The FSUtils will throw a fatal exception if the operation fails, which we don't want for the config/settings.